Source/Scene/Primitive.js

```diff
@@ -1247,6 +1269,77 @@ define([
         }
     }
 
+    function recomputeBoundingSpheres(primitive, frameState) {
+        var offsetIndex = primitive._batchTableAttributeIndices.offset;
+        if (!primitive._recomputeBoundingSpheres || !defined(offsetIndex)) {
+            return;
+        }
+
+        var i;
+        var bs;
+        var boundingSpheres = primitive._unmodifiedInstanceBoundingSpheres;
+        var newBoundingSpheres = primitive._instanceBoundingSpheres;
+        for (i = 0; i < boundingSpheres.length; ++i) {
+            bs = boundingSpheres[i];
+            var newBS = newBoundingSpheres[i];
+            var offset = primitive._batchTable.getBatchedAttribute(i, offsetIndex, new Cartesian3());
+            newBS = bs.clone(newBS);
+            transformBoundingSphere(newBS, offset, true);
+        }
+        var combinedBS = [];
+        var combinedWestBS = [];
+        var combinedEastBS = [];
+
+        for (i = 0; i < newBoundingSpheres.length; ++i) {
+            bs = newBoundingSpheres[i];
+
+            var minX = bs.center.x - bs.radius;
+            if (minX > 0 || BoundingSphere.intersectPlane(bs, Plane.ORIGIN_ZX_PLANE) !== Intersect.INTERSECTING) {
+                combinedBS.push(bs);
+            } else {
+                combinedWestBS.push(bs);
+                combinedEastBS.push(bs);
+            }
+        }
+
+        var resultBS1 = combinedBS[0];
+        var resultBS2 = combinedEastBS[0];
+        var resultBS3 = combinedWestBS[0];
+
+        for (i = 1; i < combinedBS.length; i++) {
+            resultBS1 = BoundingSphere.union(resultBS1, combinedBS[i]);
+        }
+        for (i = 1; i < combinedEastBS.length; i++) {
+            resultBS2 = BoundingSphere.union(resultBS2, combinedEastBS[i]);
+        }
+        for (i = 1; i < combinedWestBS.length; i++) {
+            resultBS3 = BoundingSphere.union(resultBS3, combinedWestBS[i]);
+        }
+        var result = [];
+        if (defined(resultBS1)) {
+            result.push(resultBS1);
+        }
+        if (defined(resultBS2)) {
+            result.push(resultBS2);
+        }
+        if (defined(resultBS3)) {
+            result.push(resultBS3);
+        }
+
+        if (result.length !== primitive._boundingSpheres.length) {
+            throw new DeveloperError('whoops');
```

Comment:
Has this ever happened?

Reply:
Not yet.  I put this in here as I was debugging just to check, but I'll remove it before merging this into master